### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by sayyed-kamran**

### CODE REVIEW

File: src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,178 @@
+use crate::{mock::*, Error};
+use frame_support::{assert_noop, assert_ok};
+
+#[test]
+fn add_item_test(){
```
Comment (sfffaaa) [inline src/tests.rs]: I saw there are several tests in one test case. If each is independent, please divide it into different test cases to improve the maintenance. For example, I will split it into four other test cases in your add test case. 
1. Add the successful case
2. Error::<Test>::ItemTypeAlreadyExists
3. Error::<Test>::ItemTypeExceedMax64
4. Error::<Test>::ItemExceedMax128

**Review by sfffaaa — CHANGES_REQUESTED** [review]

File: src/tests.rs (hunk 1 above)
Comment (sayyed-kamran) [reply src/tests.rs]: I restructured unit tests. Please review.

File: src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+    });
+}
+
+#[test]
+fn update_item_with_item_length_exceed_limit_test() {
+    new_test_ext().execute_with(|| {
+        System::set_block_number(1);     
+
+        let acct="Iredia";
```
Comment (sfffaaa) [inline src/tests.rs]: I saw the same strings are used in the codebase; could we use one variable or constant to store it?

Comment (sayyed-kamran) [reply src/tests.rs]: Moved strings used for acct and acct2 to global scope.

File: src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    });
+}
+
+#[test]
+fn update_item_with_item_length_exceed_limit_test() {
+    new_test_ext().execute_with(|| {
+        System::set_block_number(1);     
+
+        let origin = account_key(ACCT);
+        let item_type = b"itemType";
```
Comment (sfffaaa) [inline src/tests.rs]: How about the item and item_type? Should we extract this?

Comment (sayyed-kamran) [reply src/tests.rs]: At times we need to change item and item type but let me work it out and find some work around for this as well.

Comment (sayyed-kamran) [reply src/tests.rs]: Moved the item and item type to global scope as constants as well.

**Review by sfffaaa — APPROVED** [review]

**Pull request merged**